### PR TITLE
stepper component

### PR DESCRIPTION
**Pull request by HellraiserCW**
Stepper component with auth service

### CODE REVIEW

File: src/app/auth/components/stepper/stepper.component.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      <img ngSrc="/assets/icons/{{ stepperIcons.firstStepIcon }}.svg" width="40" height="40" alt="first step" />
+    </button>
+  </span>
+  <span class="stepper__divider">
+    <button class="stepper__step" [disabled]="disabled" (click)="onClick(2)">
+      <img ngSrc="/assets/icons/{{ stepperIcons.secondStepIcon }}.svg" width="40" height="40" alt="second step" />
+    </button>
+  </span>
+  <button class="stepper__step" [disabled]="disabled" (click)="onClick(3)">
+    <img ngSrc="/assets/icons/{{ stepperIcons.thirdStepIcon }}.svg" width="40" height="40" alt="third step" />
```
Comment (GennadiyShpak) [inline src/app/auth/components/stepper/stepper.component.html:13]: need more details in alt attribute

File: src/app/auth/pages/sign-up/sign-up.component.ts [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
   templateUrl: './sign-up.component.html',
   styleUrls: ['./sign-up.component.scss']
 })
-export default class SignUpComponent {}
+export default class SignUpComponent {
+  constructor(public authService: AuthService) {}
+  handleClickedStep(event: number): void {
+    this.authService.handleClickedStep(event);
+  }
+}
```
Comment (GennadiyShpak) [inline src/app/auth/pages/sign-up/sign-up.component.ts:18]: where space between rows?

File: src/app/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  stepperState: StepperState = {
+    firstStepPassed: false,
+    secondStepPassed: false,
+    thirdStepPassed: false
+  };
+  currentStepperIcons: StepperIcons = {
+    firstStepIcon: STEPPER_ICONS.FirstStepActive,
+    secondStepIcon: STEPPER_ICONS.SecondStepDisabled,
+    thirdStepIcon: STEPPER_ICONS.ThirdStepDisabled
+  };
```
Comment (GennadiyShpak) [inline src/app/services/auth.service.ts:18]: i think config.ts the better place for this initial data

File: src/app/services/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        this.stepperState.thirdStepPassed ? STEPPER_ICONS.FilledStep : STEPPER_ICONS.ThirdStepDisabled
+      );
+    } else if (this.stepperState.firstStepPassed) {
+      if (event === 2) {
+        this.stepperState.secondStepPassed = true;
+        this.handleStepperIcons(
+          STEPPER_ICONS.FilledStep,
+          STEPPER_ICONS.SecondStepActive,
+          this.stepperState.thirdStepPassed ? STEPPER_ICONS.FilledStep : STEPPER_ICONS.ThirdStepDisabled
+        );
```
Comment (GennadiyShpak) [inline src/app/services/auth.service.ts:34]: avoid nesting please

File: src/app/services/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    secondStepPassed: false,
+    thirdStepPassed: false
+  };
+  currentStepperIcons: StepperIcons = {
+    firstStepIcon: STEPPER_ICONS.FirstStepActive,
+    secondStepIcon: STEPPER_ICONS.SecondStepDisabled,
+    thirdStepIcon: STEPPER_ICONS.ThirdStepDisabled
+  };
+
+  handleClickedStep(event: number): void {
```
Comment (GennadiyShpak) [inline src/app/services/auth.service.ts:20]: need discuss this function and improve it

File: src/app/auth/pages/sign-up/sign-up.component.html [hunk 6]
```diff
@@ -1 +1,4 @@
-<p>sign-up works!</p>
+<epm-stepper />
+<epm-button class="primary" (click)="onNextStepClick()">
+  {{ authService.stepperConfig().activeStep === stepperSteps.ThirdStep ? 'Create account' : 'Continue' }}
```
Comment (GennadiyShpak) [inline src/app/auth/pages/sign-up/sign-up.component.html:3]: Can we move this logic to component.ts?

Comment (HellraiserCW) [reply src/app/auth/pages/sign-up/sign-up.component.html:3]: Logic moved to component

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**